main/svgparser.js

```diff
@@ -1572,6 +1508,85 @@ export class SvgParser {
 			if (command=='M' || command=='m') x0=x, y0=y;
 		}
 
+		// After generating the polygon points, check for self-intersection.
+		if (poly.length > 0) {
+			let uniquePolyForCheck = poly.slice();
+
+			// Remove duplicate last point if it's the same as the first (for closed paths)
+			if (uniquePolyForCheck.length > 1 && GeometryUtil.almostEqualPoints(uniquePolyForCheck[0], uniquePolyForCheck[uniquePolyForCheck.length - 1], this.conf.toleranceSvg)) {
+				uniquePolyForCheck.pop();
+			}
+
+      if (uniquePolyForCheck.length >= 3 && this.isPolygonSelfIntersecting(uniquePolyForCheck)) {
```

Comment:
1. I think this is really two error cases: less than 3 points, not a polygon we can nest and self-intersecting polygon, which should generate different error messages.
2. I don't think self-intersecting polygons should necessarily be skipped. If I want to cut out a pair of drop shapes 💧 by making a single 8, I think it's reasonable to cut that out.